test_helpers/src/lib.rs

```diff
@@ -93,6 +95,10 @@ pub fn start_logging() {
 
         let subscriber = FmtSubscriber::builder()
             .with_env_filter(EnvFilter::from_default_env())
+            // Note `with_test_writer` allows cargo-test to capture
+            // logging (which may reorder / hide input) if you don't
+            // see your expected logs, try passing `--no-capture` to
+            // `cargo test`
```

Comment:
I'm not aware of reordering (within the same test). What the capturing does is to present you the output for failed tests in a "summary". This is super handy for CI systems, but also locally when you run multiple tests (remember: test execution is multi-threaded, so you won't have a good way to disentangle direct stdouts without capturing). I agree that this may conflict w/ debugging. Also your explanation is technically a bit off (don't wanna be too picky, but it may confuse a future reader), so how about:

```suggestion
            // Note `with_test_writer` allows libtest (used for all 
            // our tests and invoked by `cargo test`) to capture
            // per-test logging. The captured data will only be 
            // shown for failed tests. Pass `--no-capture` to 
            // disable that feature (but only try to run a single 
            // test to prevent output interleaving).
```